src/cmd/services/m3coordinator/ingest/write.go

```diff
@@ -49,6 +52,8 @@ type DownsampleAndWriteIter interface {
 	Current() (models.Tags, ts.Datapoints, ts.SeriesAttributes, xtime.Unit, []byte)
 	Reset() error
 	Error() error
+	SetCurrentMetadata(ts.Metadata)
+	CurrentMetadata() ts.Metadata
```

Comment:
Would it be better to just add ts.Metadata to the return for `.Current()`?

Reply:
Yeah did that. Also pulled in all the things returned by current into a single struct..